tsfile/src/test/java/org/apache/iotdb/tsfile/write/schema/converter/SchemaBuilderTest.java

```diff
@@ -65,10 +68,13 @@ public void testJsonConverter1() {
     String[] tsDesStrings = {
       "[s4,DOUBLE,RLE,{max_point_number=3},SNAPPY]", "[s5,INT32,TS_2DIFF,,UNCOMPRESSED]"
     };
-    int i = 0;
+    Set<String> expected = new HashSet<String>(Arrays.asList(tsDesStrings));
+    List<String> actual = new ArrayList<String>();
     for (IMeasurementSchema desc : timeseries) {
-      assertEquals(tsDesStrings[i++], desc.toString());
+      actual.add(desc.toString());
     }
+    assertEquals(expected.size(), actual.size());
+    assertEquals(expected, new HashSet<String>(actual));
```

Comment:
`nondex` seems very interesting. 
However, I have a question: 
nondex is for checking `iterator of HashMap`, should we also consider the useage of `List`?
For example, in this case, `timeseres` is an `ArrayList`.
 
The second, `assertEquals(expected, new HashSet<String>(actual));` will compare two `HashSet` ( and use while-loop for checking all elements), is it ok (and is it more stable than ArrayList?)

Reply:
Thanks for your question. 
First, `nondex` doesn't shuffle lists because they have a specific iteration order. Second, the assertion compares two sets rather than two lists precisely to avoid the dependency on the order.